src/main/java/com/openshift/internal/restclient/model/KubernetesResource.java

```diff
@@ -161,20 +158,26 @@ public void addLabel(String key, String value) {
 	
 	/*---------- utility methods ------*/
 	protected ModelNode get(String key){
-		String [] property = propertyKeys.get(key);
-		return node.get(property);
+		return node.get(getPath(key));
 	}
-	
+
 	protected void set(String key, int value) {
-		String [] property = propertyKeys.get(key);
-		node.get(property).set(value);
+		node.get(getPath(key)).set(value);
 	}
 	
 	protected void set(String key, String value){
-		String [] property = propertyKeys.get(key);
-		node.get(property).set(value);
+		node.get(getPath(key)).set(value);
 	}
 	
+	private String[] getPath(String key) {
+		String [] property = propertyKeys.get(key);
+		if (property == null) {
```

Comment:
@jcantrill isnt the property map a mapping so that if a special mapping is missing it indicates that the key is to be used as is? What is the purpose of these property-keys?